src/tm2/tm-device.cpp

```diff
@@ -178,6 +178,32 @@ namespace librealsense
         tm2_sensor&         _ep;
     };
 
+    template <perc::SIXDOF_MODE flag, perc::SIXDOF_MODE depends_on>
+    class tracking_mode_option : public option_base
+    {
+    public:
+        float query() const override { return (s._tm_mode & flag) ? 1 : 0; }
+
+        void set(float value) override {
+            if (s._is_streaming)
```

Comment:
Shouldn't it be possible to setup the desired configuration before streaming  - both for convenience and also to avoid corner cases?